server/src/lib/migration/migrations/chunithm-lamp-split.ts

```diff
@@ -146,6 +146,7 @@ function MoveLamps(
 	} else if (
 		oldLamp === "FULL COMBO" ||
 		oldLamp === "ALL JUSTICE" ||
+		// eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
```

Comment:
ESLint's heuristic is poor here, but what it's saying is correct. This condition is unecessary because all other possible states have been exhausted.

With a more simple example, what you are saying is the same thing as `if (cond !== true && cond === false)`; After the previous conditions there is only one possible value it can be, so the check is redundant.

However, I am completely with you here that this check should be maintained and ESLint is wrong for flagging this as an issue. It's just a very poor quality linter.